simulator/virtual_machine_test.go

```diff
@@ -446,3 +446,89 @@ func TestShutdownGuest(t *testing.T) {
 
 	}
 }
+
+func TestVmSnapshot(t *testing.T) {
+	ctx := context.Background()
+
+	m := ESX()
+	defer m.Remove()
+	err := m.Create()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := m.Service.NewServer()
+	defer s.Close()
+
+	c, err := govmomi.NewClient(ctx, s.URL, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	finder := find.NewFinder(c.Client, false)
+	finder.SetDatacenter(object.NewDatacenter(c.Client, esx.Datacenter.Reference()))
+
+	vms, err := finder.VirtualMachineList(ctx, "*")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	vm := vms[0]
```

Comment:
Note that for the test cases where you can use any instead of a type, you can avoid the Finder hassle with the `Map.Any` function:

```go
vm := object.NewVirtualMachine(c, Map.Any("VirtualMachine").Reference())
```